src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -77,11 +77,24 @@ public void visitToken(DetailAST ast) {
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!containCaseBeforeOrAfterDefault(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+                        || defaultGroupAST.getColumnNo()
+                            != lastGroupAST.getColumnNo())) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * Examines ast to see if it is surrounded by a 'case' on either side.
+     *
+     * @param ast - root node.
+     * @return true if default is surrounded by a 'case' on either side or false.
+     */
+    private boolean containCaseBeforeOrAfterDefault(DetailAST ast) {
+        return ast.getPreviousSibling() != null
+                || ast.getNextSibling() != null
+                && ast.getNextSibling().getType() == TokenTypes.LITERAL_CASE;
```

Comment:
> comments are skipped in AST for java Checks at all, so it should not be a problem.

Good to know, thanks! Also according to the [java syntax](https://docs.oracle.com/javase/specs/jls/se7/html/jls-18.html), only `case` and `default` are siblings.
```java
case Expression :
case EnumConstantName :
default :
```
So ignore my comment.

> if you have code with default not in the end in your project - please share a link to it to let us verify our implementation on real code.

Unfortunately I can't provide [the project I'm working on](https://github.com/Catrobat/Catroid) because we are using the `DefaultComesLast` check. :disappointed: